src/main/kotlin/test/ktortemplate/core/service/CarServiceImpl.kt

```diff
@@ -18,7 +19,7 @@ class CarServiceImpl : KoinComponent, CarService {
     private val dbc: DatabaseConnection by inject()
 
     override suspend fun count(pageRequest: PageRequest): Int {
-        return dbc.suspendedQuery { carRepository.count(pageRequest) }
+        return dbc.suspendedQueryContext(MDCContext()) { carRepository.count(pageRequest) }
```

Comment:
Here is the new function I've created, where I pass the MDC context that contains the request id that was loaded from the **X-Request-Id** header. This way the new coroutine contains this context and can present the requested id that originated the log call.

Reply:
Can we provide these `MDCContext` in the `DatabaseConnection`? Just to clear up this responsibility on this service?

Reply:
If we use logs inside the `suspendedQuery` block and want them to present the caller request id we have to pass the `MDCContext`. But we can come to the conclusion that logs are not important in this context and assume that logs inside this block will always be stripped of the request id.

Reply:
I simplified this a bit, as I don't think the client should have to deal with the boilerplate of creating a new MDCContext